integration/helpers.go

```diff
@@ -203,3 +203,39 @@ func AssertEqualWriteError(t *testing.T, expected *mongo.WriteError, actual erro
 	return assert.Equal(t, expected.Message, actualWriteErr.Message) &&
 		assert.Equal(t, expected.Code, actualWriteErr.Code)
 }
+
+// AssertEqualAltWriteError compares expected mongo.WriteError Message and Code with actual error.
+func AssertEqualAltWriteError(t *testing.T, expected *mongo.WriteError, alt string, actual error) bool {
+	t.Helper()
+
+	var actualWE mongo.WriteError
+	actualEx, ok := actual.(mongo.WriteException)
+	if ok {
+		if len(actualEx.WriteErrors) != 1 {
+			return assert.Equal(t, expected, actual)
+		}
+		actualWE = actualEx.WriteErrors[0]
+	} else {
```

Comment:
I would expect `AssertEqualAltWriteError` to do the same as `AssertEqualWriteError` with additional check for alternative error message. But it does much more: checks for both WriteException and WriteError, logs a lot, etc. I think it should not do all that.

Reply:
Would you please propose your solution in that case?

Reply:
I propose to remove checking for error type that is not returned by the driver and logging. I propose to make this function just like AssertEqualWriteError with an additional check for the alternative error message and remove everything unrelated to it.

Reply:
`Array` and `NilOperand` returning `mongo.WriteException`

Reply:
@AlekSi  I've changed the function, but it still can accept `mongo.Exception` and `mongo.WriteError`.

Reply:
// WriteError is an error that occurred during execution of a write operation.
// **This error type is only returned as part of a WriteException or BulkWriteException.**

Reply:
> But it does much more: checks for both WriteException and WriteError

> propose to remove checking for error type that is not returned by the driver

> `Array` and `NilOperand` returning `mongo.WriteException`

Ok, what about the second type? Let's check documentation:

> // WriteError is an error that occurred during execution of a write operation.
> // **This error type is only returned as part of a WriteException or BulkWriteException.**

`actual` can never be `WriteError` because that error is never returned by the driver.

Reply:
removed, renamed. PTAL.